src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -538,6 +544,51 @@ getModule df n = do
 
 -- ---------------------------------------------------------------------
 
+-- | Return the type definition of the symbol at the given position.
+-- Works for Datatypes, Newtypes and Type Definitions, as well as paremterized types.
+-- Type Definitions can only be looked up, if the corresponding type is defined in the project.
+-- Sum Types can also be searched.
+findTypeDef :: Uri -> Position -> IdeDeferM (IdeResult [Location])
+findTypeDef uri pos = pluginGetFile "findTypeDef: " uri $ \file ->
+  ifCachedInfo
```

Comment:
@mpickering @fendor I actually think `withCachedInfo` is suitable and possibly preferred here: This is the type of request explicitly requested by the user and so I think it would make sense for hie to wait until it has typechecked the module for it rather than just returning nothing